### PR TITLE
Upgrading to Swift 4 and Xcode 9

### PR DESCRIPTION
**Pull request by bsrz**
- Upgrading to recommended settings (Xcode 9)
- Upgrading to latest Swift syntax (Swift 4)
- Removing redundant warnings

### CODE REVIEW

File: RxRealm.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -485,7 +505,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = com.RxSwiftCommunity.RxRealm;
 				PRODUCT_NAME = RxRealm;
 				SKIP_INSTALL = YES;
-				SWIFT_VERSION = 3.0;
+				SWIFT_VERSION = 4.0;
```
Comment (ajjames) [inline RxRealm.xcodeproj/project.pbxproj]: 🕺

**Review by ajjames — APPROVED** [review]

**Comment by icanzilb** [comment]
is that different than the current swift4 branch?

**Comment by bsrz** [comment]
@icanzilb it's different in the sense that it doesn't require beta versions of dependencies.
Do you know what is the ETA for releasing RxSwift 4 and Realm 3?

**Comment by icanzilb** [comment]
It looks like the current betas of both RxSwift and RealmSwift will be the last ones before a release, but I do not know that for sure.

**Comment by icanzilb** [comment]
ok, I see - this PR looks to be dealing mostly with Carthage stuff, I think we can safely merge it to `master`

**Comment by bsrz** [comment]
@icanzilb yes, it fixes https://github.com/RxSwiftCommunity/RxRealm/issues/86 and there were no code changes from Swift 3 to Swift 4. I was going to try setting `SWIFT_VERSION` to 3 to validate that all works. Let me know what you want to do.

**Comment by bsrz** [comment]
@icanzilb I have confirmed all is working with Xcode 9 and Swift 3, feel free to merge this in and we'll wait for the swift4 branch on our side.

**Comment by bsrz** [comment]
@icanzilb do you still intend on merging this and tag as 0.7.2?
Just trying to point to this repo instead of my fork.

**Comment by icanzilb** [comment]
@bsarrazin yeah, will merge this today

**Pull request merged**